app/models/candidate_interface/satisfaction_survey_form.rb

```diff
@@ -4,12 +4,35 @@ class SatisfactionSurveyForm
 
     attr_accessor :question, :answer
 
+    QUESTIONS_WE_ASK = [
+      'I would recommend this service to a friend or colleague',
+      'I found this service unnecessarily complex',
+    ].freeze
+
     validates :question, presence: true
+    validates :question, inclusion: { in: QUESTIONS_WE_ASK, allow_blank: false }
 
     def save(application_form)
       return false unless valid?
 
-      application_form.update(satisfaction_survey: { @question => @answer })
+      if question_already_answered?(application_form)
+        application_form.satisfaction_survey[@question] = @answer
+        application_form.save
+      elsif application_form.satisfaction_survey.present?
+        application_form.update!(satisfaction_survey: merge_satisfaction_survey_and_answer(application_form))
+      else
+        application_form.update!(satisfaction_survey: { @question => @answer })
+      end
```

Comment:
Would this work for all the cases?

```suggestion
      application_form.satisfaction_survey ||= {}
      application_form.satisfaction_survey[@question] = @answer
      application_form.save
```

Reply:
Oh, nice! Very elegant.